ethicml/algorithms/inprocess/in_algorithm.py

```diff
@@ -15,10 +17,32 @@
 class InAlgorithm(Algorithm):
     """Abstract Base Class for algorithms that run in the middle of the pipeline."""
 
-    def __init__(self, name: str, is_fairness_algo: bool = True):
-        super().__init__(name=name)
+    def __init__(self, name: str, seed: int, is_fairness_algo: bool = True):
+        super().__init__(name=name, seed=seed)
         self.__is_fairness_algo = is_fairness_algo
 
+    @abstractmethod
+    def fit(self, train: DataTuple) -> InAlgorithm:
```

Comment:
This would be the perfect opportunity to use `-> Self` but it's not widely supported yet: https://github.com/python/typeshed/issues/6300